src/mame/drivers/ibmpcjr.cpp

```diff
@@ -668,13 +666,11 @@ MACHINE_CONFIG_START(pcjr_state::ibmpcjr)
 	m_keyboard->keypress().set(FUNC(pcjr_state::keyb_interrupt));
 
 	/* cartridge */
-	MCFG_GENERIC_CARTSLOT_ADD("cartslot1", generic_plain_slot, "ibmpcjr_cart")
-	MCFG_GENERIC_EXTENSIONS("bin,jrc")
-	MCFG_GENERIC_LOAD(pcjr_state, pcjr_cart1)
+	GENERIC_CARTSLOT(config, m_cart1, generic_plain_slot, "ibmpcjr_cart", "bin,jrc");
+	m_cart1->set_device_load(device_image_load_delegate(&pcjr_state::device_image_load_pcjr_cart1, this));
```

Comment:
This is what @cuavas means when talking about expanding macros. IMHO MCFG_GENERIC_LOAD and MCFG_GENERIC_UNLOAD are better left alone until a better solution is offered.